Segmentation fault when building Coq on Windows 10

#### Version

The Coq Proof Assistant, version 8.7.0 (November 2017)
compiled on Nov 26 2017 3:03:40 with OCaml 4.06.0
[I checked out v8.7.0 in git]

#### Operating system

Windows 10
i7-5500U processor (2 core, hyperthreaded)

#### Description of the problem

When I run 'make' to build Coq, I get a Segmentation Fault every time in theories/FSets/FMapFullAVL.v:

```
$ make VERBOSE=1
make --warn-undefined-variable --no-builtin-rules -f Makefile.build
make[1]: Entering directory '/cygdrive/c/Users/Jim.DESKTOP-MMPNVGH/Documents/coqSource/coq'
rm -f theories/FSets/FMapFullAVL.glob
bin/coqtop.exe -boot   -compile theories/FSets/FMapFullAVL.v
make[1]: *** [Makefile.build:745: theories/FSets/FMapFullAVL.vo] Segmentation fault
make[1]: *** [theories/FSets/FMapFullAVL.vo] Deleting file 'theories/FSets/FMapFullAVL.glob'
make[1]: Leaving directory '/cygdrive/c/Users/Jim.DESKTOP-MMPNVGH/Documents/coqSource/coq'
make: *** [Makefile:159: submake] Error 2
```

It blows up on line 70 (or just after line 70) in that file:
```
bin/coqtop.exe -boot   -compile-verbose theories/FSets/FMapFullAVL.v -time
     :
inv avl; intuition; omega_max.
Chars 2124 - 2154 [(inv~avl;~intuition;~omega_max).] Segmentation fault
```

Emilio suggested this may be the same bug as https://github.com/coq/coq/issues/5517.  However, my processor is an i7-5500U, in the Broadwell family while that bug only mentions the Kaby Lake and Skylake families.  Broadwell is the prior family.  I've not found anything about a problem in Broadwell or a microcode update from Intel for it.

"-native-compiler no" on the ./configure didn't help

I've not found a way to turn off hyperthreading for Windows 10.

I had to convert ./configure from DOS to Unix line endings.  Also kernel/copcodes.ml had a number of return characters that made that file not compile.  I removed those.  No other changes.

This is my first time trying to compile Coq and my first use of OCaml.

Here is my ./configure:

```
$ ./configure -native-compiler no                                                                                     
You have OCaml 4.06.0. Good!
You have Camlp5 7.03. Good!
You have native-code compilation. Good!
No such directory 'C:/OCaml64/home/Jim/.opam/4.06.0+mingw64c/lib/ocaml/lablgtk2' (in OCaml library).
LablGtk2 not found:
=> no CoqIde will be built.
Where should I install the Coq binaries [C:/coq/bin]?
Where should I install the Coq library [C:/coq/lib]?
Where should I install the Coqide configuration files [C:/coq/config]?
Where should I install the Coqide data files [C:/coq/share]?
Where should I install the Coq man pages [C:/coq/man]?
Where should I install the Coq documentation [C:/coq/doc]?
Where should I install the Coq Emacs mode [C:/coq/emacs]?
Where should I install the Coqdoc LaTeX files [C:/coq/latex]?

  Architecture                : win32
  Operating system            : Windows_NT
  Coq VM bytecode link flags  : -custom
  Other bytecode link flags   : -custom
  OS dependent libraries      : -cclib -lunix
  OCaml version               : 4.06.0
  OCaml binaries in           : C:/OCaml64/home/Jim/.opam/4.06.0+mingw64c/bin/
  OCaml library in            : C:/OCaml64/home/Jim/.opam/4.06.0+mingw64c/lib/ocaml
  OCaml flambda flags         :
  Camlp5 version              : 7.03
  Camlp5 binaries in          : C:\OCaml64\home\Jim\.opam\4.06.0+mingw64c\bin
  Camlp5 library in           : C:/OCaml64/home/Jim/.opam/4.06.0+mingw64c/lib/camlp5
  Native dynamic link support : true
  CoqIde                      : no
  Documentation               : None
  Web browser                 : start %s
  Coq web site                : http://coq.inria.fr/

  Native compiler for conversion and normalization disabled

  Paths for true installation:
  - the Coq binaries will be copied in C:/coq/bin
  - the Coq library will be copied in C:/coq/lib
  - the Coqide configuration files will be copied in C:/coq/config
  - the Coqide data files will be copied in C:/coq/share
  - the Coq man pages will be copied in C:/coq/man
  - the Coq documentation will be copied in C:/coq/doc
  - the Coq Emacs mode will be copied in C:/coq/emacs
  - the Coqdoc LaTeX files will be copied in C:/coq/latex

If anything is wrong above, please restart './configure'.

*Warning* To compile the system for a new architecture
          don't forget to do a 'make clean' before './configure'.
```

<!-- It is helpful to provide enough information so that we can reproduce the bug.
     In particular, please include a code example which produces it.
     If the example is small, you can include it here between ``` ```.
     Otherwise, please provide a link to a repository, a gist (https://gist.github.com)
     or drag-and-drop a `.zip` archive. -->

--- a/Makefile.build
+++ b/Makefile.build
@@ -302,7 +302,7 @@ kernel/byterun/coq_jumptbl.h : kernel/byterun/coq_instruct.h
                -e '/^}/q' $< $(TOTARGET)
 
 kernel/copcodes.ml: kernel/byterun/coq_instruct.h
-	sed -n -e '/^enum/p' -e 's/,//g' -e '/^  /p' $< | \
+	tr -d "\r" < $< | sed -n -e '/^enum/p' -e 's/,//g' -e '/^  /p' | \
 	awk -f kernel/make-opcodes $(TOTARGET)
 
 %.o: %.c

--- a/Makefile.checker
+++ b/Makefile.checker
@@ -75,8 +75,9 @@ checker/%.cmx: checker/%.ml
 
 md5chk:
 	$(SHOW)'MD5SUM cic.mli'
-	$(HIDE)if grep -q `$(MD5SUM) checker/cic.mli` checker/values.ml; \
-	       then true; else echo "Error: outdated checker/values.ml"; false; fi
+	$(HIDE)v=`tr -d "\r" < checker/cic.mli | $(MD5SUM) | sed -n -e 's/ .*//' -e '/^/p'`; \
+	       if grep -q "$$v" checker/values.ml; \
+	       then true; else echo "Error: outdated checker/values.ml: $$v" >&2; false; fi
 
 .PHONY: md5chk
 

--- a/dev/build/windows/MakeCoq_MinGW.bat
+++ b/dev/build/windows/MakeCoq_MinGW.bat
@@ -345,7 +345,7 @@ IF "%COQREGTESTING%" == "Y" (
 SET "EXTRAPACKAGES= "
 
 IF NOT "%APPVEYOR%" == "True" (
-  SET EXTRAPACKAGES="-P wget,curl,git,gcc-core,gcc-g++,automake1.5"
+  SET EXTRAPACKAGES=-P wget,curl,git,gcc-core,gcc-g++,automake1.5
 )
 
 IF "%RUNSETUP%"=="Y" (

